contracts/Delegator.sol

```diff
@@ -27,10 +45,35 @@ contract Delegator is ACL {
             string memory url
         )
     {
-        return assetManager.getJob(id);
+        return assetManager.getJob(ids[_name]);
     }
 
-    // function getResult(uint8 id) public view returns (uint256) {
-    //     return assetManager.getResult(id);
-    // }
+    function getCollection(bytes32 _name)
+        external
+        view
+        returns (
+            bool active,
+            int8 power,
+            uint8[] memory jobIDs,
+            uint32 aggregationMethod,
+            string memory name
+        )
+    {
+        return assetManager.getCollection(ids[_name]);
+    }
+
+    function getNumActiveAssets() external view returns (uint256) {
+        return (assetManager.getActiveAssets()).length;
+    }
+
+    function getActiveAssets() external view returns (uint8[] memory) {
+        return assetManager.getActiveAssets();
+    }
+
+    function getResult(bytes32 _name) public view returns (uint32) {
+        uint8 index = assetManager.getAssetIndex(ids[_name]);
+        uint32 epoch = parameters.getEpoch();
+        uint32[] memory medians = blockManager.getBlockMedians(epoch - 1);
```

Comment:
As its external call, maybe we can optimise this to only fetch medians for given index, not for all and then pick one